pkg/services/alerting/conditions/evaluator.go

```diff
@@ -6,20 +6,27 @@ import (
 	"github.com/grafana/grafana/pkg/components/null"
 	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/services/alerting"
+	"github.com/grafana/grafana/pkg/tsdb"
 )
 
 var (
 	defaultTypes []string = []string{"gt", "lt"}
+	queryTypes   []string = []string{"above_metric", "below_metric"}
 	rangedTypes  []string = []string{"within_range", "outside_range"}
 )
 
 type AlertEvaluator interface {
-	Eval(reducedValue null.Float) bool
+	Update(ctx *alerting.EvalContext) error
```

Comment:
My initial implementation did without, but then I was re-issuing the comparison query for every series returned by the initial query.  Alternatively, the `Eval` interface could be rewritten such that it takes the full result set and a reducer, returning an EvalMatch struct. That seemed a more destructive change, so I avoided it at the time